src/main/java/com/sprint_manager/repos/SprintRepo.java

```
@@ -0,0 +1,106 @@
+        );
+    }
+
+    public Integer insertNewTask(Sprint sprint) {
+        return jdbcTemplate.queryForObject(
+                "insert into sprints (title, author_id, state, date_start, date_end_expect) " +
+                        "values(?, ?, ?, ?, ?) returning id",
+                new Object[] {
+                        sprint.getTitle(),
+                        sprint.getAuthorId(),
+                        sprint.getSprintState(),
+                        sprint.getStartDate(),
+                        sprint.getEndDateExpect()
+                },
+                Integer.class
+        );
+    }
+
+    public Sprint getSprintById(Integer id) {
+        return jdbcTemplate.queryForObject(
+                "select s.id, u.full_name, s.title, s.state, " +
+                        "s.date_start, s.date_end_expect, s.date_end_fact " +
+                        "from sprints as s inner join users as u on s.author_id = u.id " +
+                        "where s.id = ?",
+                new Object[] {id},
+                new SprintMapper()
+        );
+    }
+
+    public void updateSprintStateAndActEndDate(Integer id, String state, Date dateEndFact) {
+        jdbcTemplate.update(
+                "update sprints set state = ?, date_end_fact = ? where id = ?",
+                state,
+                dateEndFact,
+                id
+        );
+    }
+
+    public Sprint getActiveSprint() {
+        try {
+            return jdbcTemplate.queryForObject(
+                    "select s.id, u.full_name, s.title, s.state, " +
+                            "s.date_start, s.date_end_expect, s.date_end_fact " +
+                            "from sprints as s inner join users as u on s.author_id = u.id " +
+                            "where s.state = ?",
+                    new Object[] { SprintState.STARTED.toString() },
+                    new SprintMapper()
+            );
+        } catch (IncorrectResultSizeDataAccessException e) {
```

Please, explain why you are catching `IncorrectResultSizeDataAccessException ` ?

---

`IncorrectResultSizeDataAccessException `- if the query does not return exactly one row, or does not return exactly one column in that row

---

> Please, explain why you are catching `IncorrectResultSizeDataAccessException ` ?

`IncorrectResultSizeDataAccessException `- if the query does not return exactly one row, or does not return exactly one column in that row

---

To my mind it'd be better to re-throw this exception.